diff -Nru tree/etc/gdm/gdm.conf ubuntu-8.04-tree/etc/gdm/gdm.conf
--- tree/etc/gdm/gdm.conf	2008-07-10 10:55:06.000000000 -0300
+++ ubuntu-8.04-tree/etc/gdm/gdm.conf	2008-07-10 10:55:26.000000000 -0300
@@ -43,13 +43,19 @@
 DynamicXServers=true
 FlexibleXServers=0
 
+Greeter=/usr/lib/gdmgreeter
+
 [security]
 
 [xdmcp]
 
 [gui]
+GtkTheme=Human
 
 [greeter]
+BackgroundColor=#dab082
+GraphicalThemedColor=#dab082
+GraphicalTheme=Human
 
 [chooser]
 
diff -Nru tree/etc/gdm/gdm.conf-custom ubuntu-8.04-tree/etc/gdm/gdm.conf-custom
--- tree/etc/gdm/gdm.conf-custom	2008-07-10 10:55:06.000000000 -0300
+++ ubuntu-8.04-tree/etc/gdm/gdm.conf-custom	2008-07-10 10:55:36.000000000 -0300
@@ -43,13 +43,19 @@
 DynamicXServers=true
 FlexibleXServers=0
 
+Greeter=/usr/lib/gdmgreeter
+
 [security]
 
 [xdmcp]
 
 [gui]
+GtkTheme=Human
 
 [greeter]
+BackgroundColor=#dab082
+GraphicalThemedColor=#dab082
+GraphicalTheme=Human
 
 [chooser]
 
