diff -Nru X11/xkb/keycodes/evdev X11.patched/xkb/keycodes/evdev
--- X11/xkb/keycodes/evdev	2008-09-19 11:55:04.000000000 -0300
+++ X11.patched/xkb/keycodes/evdev	2008-09-19 11:54:49.000000000 -0300
@@ -147,14 +147,16 @@
 
 	<HZTG> =  93;	// Hankaku/Zenkakau toggle
 	<HKTG> = 101;	// Hiragana/Katakana toggle
-	<AB11> = 211;	// backslash/underscore
+	//<AB11> = 211;	// backslash/underscore
 	<HENK> = 100;	// Henkan
 	<MUHE> = 102;	// Muhenkan
 	<AE13> = 132;	// Yen
 	<KATA> =  98;	// Katakana
 	<HIRA> =  99;	// Hiragana
 	<JPCM> = 103;	// KPJPComma
-	<RO>   =  97;	// Romaji
+	//<RO>   =  97;	// Romaji
+	// Fix for ABNT2 keyboards:
+	<AB11> = 97;
 
 	// Keys that are generated on Korean keyboards
 
diff -Nru X11/XKeysymDB X11.patched/XKeysymDB
--- X11/XKeysymDB	2008-09-19 11:55:04.000000000 -0300
+++ X11.patched/XKeysymDB	2008-09-19 11:54:50.000000000 -0300
@@ -251,7 +251,7 @@
 XF86Launch1		:1008FF41
 XF86Launch2		:1008FF42
 XF86Launch3		:1008FF43
-XF86Launch4		:1008FF44
+//XF86Launch4		:1008FF44
 XF86Launch5		:1008FF45
 XF86Launch6		:1008FF46
 XF86Launch7		:1008FF47
diff -Nru X11/XKeysymDB.orig X11.patched/XKeysymDB.orig
--- X11/XKeysymDB.orig	2008-09-19 11:54:36.000000000 -0300
+++ X11.patched/XKeysymDB.orig	2008-09-19 11:54:48.000000000 -0300
@@ -251,7 +251,7 @@
 XF86Launch1		:1008FF41
 XF86Launch2		:1008FF42
 XF86Launch3		:1008FF43
-//XF86Launch4		:1008FF44
+XF86Launch4		:1008FF44
 XF86Launch5		:1008FF45
 XF86Launch6		:1008FF46
 XF86Launch7		:1008FF47
